wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -62,6 +68,24 @@ class PlatformRepo(val walletApplication: WalletApplication) {
 
     companion object {
         private val log = LoggerFactory.getLogger(PlatformRepo::class.java)
+
+        private val onContactsUpdatedListeners = arrayListOf<OnContactsUpdated>()
+        private val updatingContacts = AtomicBoolean(false)
+
+        fun addContactsUpdatedListener(listener: OnContactsUpdated) {
+            onContactsUpdatedListeners.add(listener)
+        }
+
+        fun removeContactsUpdatedListener(listener: OnContactsUpdated?) {
+            onContactsUpdatedListeners.remove(listener)
+        }
+
+        fun queueContactsUpdatedListeners(updateBackgroundHandler: Handler) {
+            updateBackgroundHandler.post {
+                for (listener in onContactsUpdatedListeners)
+                    listener.onContactsUpdated()
+            }
+        }
```

Comment:
This section of code was made "static" because there could be more than one instance of PlatformRepo running in the app.  DashPayViewModel has one instance.  The BlockchainServiceImpl creates a new instance every time the timer triggers an update of the DashPay state (currently only updates the contact requests and profiles)

Reply:
@HashEngineering I think we should convert PlatformRepo to the Singleton.